src/Stubs/EloquentBuilder.stubphp

```diff
@@ -353,12 +372,662 @@ class Builder
 
     /**
      * @param TModel $model
-     * @return self<TModel>
+     * @return static
      */
     public function setModel(Model $model) { }
-
+    
     /**
      * @return TModel|null
      */
     public function first($columns = ['*']) { }
+
+    /**
+     * @param  array|mixed  $columns
+     * @return static
+     */
+    public function select($columns = ['*']) { }
+
+    /**
+     * @param  \Closure|$this|string  $query
+     * @param  string  $as
+     * @return static
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function selectSub($query, $as) { }
+
+    /**
+     * @param  string  $expression
+     * @param  array  $bindings
+     * @return static
+     */
+    public function selectRaw($expression, array $bindings = []) { }
+
+    /**
+     * @param  \Closure|\Illuminate\Database\Query\Builder|string  $query
+     * @param  string  $as
+     * @return static
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function fromSub($query, $as) { }
+
+
+    /**
+     * @param  string  $expression
+     * @param  mixed  $bindings
+     * @return static
+     */
+    public function fromRaw($expression, $bindings = []) { }
+
+    /**
+     * @param  array|mixed  $column
+     * @return static
+     */
+    public function addSelect($column) { }
+
+    /**
+     * @return static
+     */
+    public function distinct() { }
+
+    /**
+     * @param  \Closure|\Illuminate\Database\Query\Builder|string  $table
+     * @param  string|null  $as
+     * @return static
+     */
+    public function from($table, $as = null) { }
```

Comment:
Could you help me understand why all these methods are needed? Maybe a feature test if you wouldn't mind?

Reply:
These methods are from Query/Builder. I tried to add them to QueryBuilder.stubphp but it doesn't work. If I don't specify them here, Psalm doesn't infer correct type.

Case 1
`Comment::query()->whereDate('created_at', '>=', Carbon::now())`

Without whereDate in any of the stubphp:
`\Illuminate\Database\Eloquent\Relations\BelongsTo<Comment>|Illuminate\Database\Query\Builder`

With whereDate defined in the **Query**Builder.stubphp; OR with whereDate defined in the **Eloquent**Builder.stubphp:
`\Illuminate\Database\Eloquent\Relations\BelongsTo<Comment>` - ok


Case 2
`$comment->article()->whereDate('created_at', '>=', Carbon::now())`

Without whereDate in any of the stubphp; OR with whereDate defined in the **Query**Builder.stubphp:
`mixed` and also error on the >= `Argument 2 of Illuminate\Database\Eloquent\Builder::__call expects array<array-key, mixed>, string(>=) providedPsalm(https://psalm.dev/004)`.

With whereDate defined in the **Eloquent**Builder.stubphp I get:
`\Illuminate\Database\Eloquent\Relations\BelongsTo<Article>` - ok


Okay, I can add some feature tests

Reply:
Interesting -- definitely something going wrong here. Let me know if you would like a hand in implementing a fix or diagnosing further.

Reply:
@mr-feek I tried to run the acceptance tests, but I could not get it working. I did find the test that should fail: "Eloquent Relation Types", the "Scenario: Relationships return themselves when the proxied method is a query builder method". 
I get these 3 errors with ErrorLevel 1:
on psalm-return in the comment:
```
Could not verify return type 'Illuminate\Database\Eloquent\Relations\HasOne<Tests\Psalm\LaravelPlugin\Models\Phone>' for App\Http\Controllers\Admin\AdvancementsController::testPsalm(https://psalm.dev/047)
```
on the code in return statement:
```Could not infer a return typePsalm(https://psalm.dev/138)```
And when I hover over the ASC:
```
Argument 2 of Illuminate\Database\Eloquent\Builder::__call expects array<array-key, mixed>, string(ASC) providedPsalm(https://psalm.dev/004)
```

I can fix it only by adding orderBy definition into EloquentBuilder stub.
Are you able to repeat this?